src/app_service/service/profile/service.nim

```diff
@@ -1,80 +1,147 @@
-      return
-    singletonInstance.userProfile.setLargeImage("")
-    singletonInstance.userProfile.setThumbnailImage("")
+      for img in response.result:
+        let imageDto = toImage(img)
+        result.add(imageDto)
+        if(imageDto.imgType == "large"):
+          singletonInstance.userProfile.setLargeImage(imageDto.uri)
+        elif(imageDto.imgType == "thumbnail"):
+          singletonInstance.userProfile.setThumbnailImage(imageDto.uri)
 
-  except Exception as e:
-    error "error: ", procName="deleteIdentityImage", errName = e.name, errDesription = e.msg
+    except Exception as e:
+      error "error: ", procName="storeIdentityImage", errName = e.name, errDesription = e.msg
 
-proc setDisplayName*(self: Service, displayName: string) =
-  try:
-    let response =  status_accounts.setDisplayName(displayName)
-    if(not response.error.isNil):
-      error "could not set display name"
-      return
-    if(not self.settingsService.saveDisplayName(displayName)):
-      error "could save display name to the settings"
+  proc deleteIdentityImage*(self: Service, address: string) =
+    try:
+      let response = status_accounts.deleteIdentityImage(address)
+      if(not response.error.isNil):
+        error "could not delete identity images"
+        return
+      singletonInstance.userProfile.setLargeImage("")
+      singletonInstance.userProfile.setThumbnailImage("")
+
+    except Exception as e:
+      error "error: ", procName="deleteIdentityImage", errName = e.name, errDesription = e.msg
+
+  proc setDisplayName*(self: Service, displayName: string) =
+    try:
+      let response =  status_accounts.setDisplayName(displayName)
+      if(not response.error.isNil):
+        error "could not set display name"
+        return
+      if(not self.settingsService.saveDisplayName(displayName)):
+        error "could save display name to the settings"
+        return
+    except Exception as e:
+      error "error: ", procName="setDisplayName", errName = e.name, errDesription = e.msg
+
+  proc requestProfileShowcasePreferences*(self: Service) =
+    try:
+      let arg = QObjectTaskArg(
+        tptr: cast[ByteAddress](asyncGetProfileShowcasePreferencesTask),
+        vptr: cast[ByteAddress](self.vptr),
+        slot: "asyncProfileShowcaseLoaded",
+      )
+      self.threadpool.start(arg)
+    except Exception as e:
+      error "Error requesting profile showcase preferences", msg = e.msg
+
+  proc asyncProfileShowcaseLoaded*(self: Service, rpcResponse: string) {.slot.} =
+    let rpcResponseObj = rpcResponse.parseJson
+    if rpcResponseObj{"error"}.kind != JNull and rpcResponseObj{"error"}.getStr != "":
+      error "Error requesting profile showcase preferences", msg = rpcResponseObj{"error"}
       return
-  except Exception as e:
-    error "error: ", procName="setDisplayName", errName = e.name, errDesription = e.msg
\ No newline at end of file
+
+    let result =  rpcResponseObj{"response"}{"result"}
```

Comment:
Using `{}` means that the property might not be defined, but since we use the property below, it has to be defined, so might as well make it clear by using `[]`
```suggestion
    let result =  rpcResponseObj["response"]["result"]
```

Reply:
fixed